apps/yggdrasil_web/test/controllers/character_controller_test.exs

```diff
@@ -0,0 +1,147 @@
+    end
+
+    # set of games to have something to list
+    games = Enum.map 1..2, fn n ->
+      game = Game.changeset(%Game{}, %{"name" => "game#{n}",
+                                       "description" => "game desc #{n}"})
+      {:ok, game} = Repo.insert game
+
+      game
+    end
+
+    users =  Enum.map users, fn u ->
+      chars = Enum.flat_map games, fn g ->
+        Enum.map 1..2, fn n ->
+          char = Character.changeset(%Character{}, %{:name => "char #{g.name} #{u.model.id} #{n}",
+                                                     :game_id => g.id,
+                                                     :user_id => u.model.id})
+
+          {:ok, char} = Repo.insert char
+
+          char
+        end
+      end
+
+      Map.put u, :chars, chars
+    end
+
+    conn = conn
+      |> put_req_header("content-type", @json_api_content_type)
+      |> put_req_header("accept", @json_api_content_type)
+
+    {:ok, %{conn: conn, games: games, users: users}}
+  end
+
+  test "get /api/characters/char_id returns the correct character", ctx do
+    user = Enum.at ctx.users, 0
+    char = Enum.at user.chars, 0
+
+    char_json = CharacterView.render("show.json", conn: %{}, data: char)
+    |> Poison.encode!
+    |> Poison.decode!
+
+    assert user.model.id == char_json["data"]["attributes"]["user-id"]
+
+    path = api_character_path(conn, :show, char.id)
+
+    conn = ctx.conn
+    |> put_req_header("authorization", user.token)
+    |> get(path)
+
+    assert response_content_type(conn, :json) == @json_api_content_type_utf8
+    resp = Poison.decode! response(conn, :ok)
+
+    assert resp == char_json
+  end
+
+  test "get /characters returns all characters for the user the token specifies", ctx do
+    user = Enum.at ctx.users, 0 # first user has the token
+    chars = Enum.sort_by user.chars, fn c -> c.id end
+
+    chars_decoded = CharacterView.render("index.json", conn: %{}, data: chars)
+    |> Poison.encode!
+    |> Poison.decode!
+
+    # assert all chracters stored in our context have the correct user_id
+    assert Enum.all? chars_decoded["data"], fn cd -> user.model.id == cd["attributes"]["user-id"] end
+
+    path = api_character_path(conn, :index)
+
+    conn = ctx.conn
+    |> put_req_header("authorization", user.token)
+    |> get(path)
+
+    assert response_content_type(conn, :json) == @json_api_content_type_utf8
+    resp = Poison.decode! response(conn, :ok)
+    # sort by id here as well.
+    resp = %{ resp | "data" => Enum.sort_by(resp["data"], fn d -> d["id"] end) }
+
+    # since all of our characters from context have the right user_id
+    # if this succeeds then all the ones from the response also have the right user_id
+    assert chars_decoded == resp
+  end
+
+  test "post /characters creates a character for a given user", ctx do
+    user = Enum.at ctx.users, 0
+    game = Enum.at ctx.games, 0
+    char = %{"name" => "unique_tester", "game-id" => game.id} # server fills in user_id
```

Comment:
Server knows what user is posting this, so it assumes the user_id needs to be that and just clobbers what the client sent and sets it on the server. I also need another test to prove this out btw.

The obvious concern here is an admin wanted to create a character for a user he can't. The counter to that, is that we don't want a user creating characters for other users.

The larger question here, which also applies to the other actions in this /characters endpoint in general is should we have two? I'll point this out in the character file may be more clear there.